smithy-build/src/main/java/software/amazon/smithy/build/FileManifest.java

```diff
@@ -69,6 +70,34 @@ static FileManifest create(Path basePath) {
     @SuppressWarnings("unused")
     Path addFile(Path path);
 
+    /**
+     * Adds the files from another FileManifest into this FileManifest.
+     *
+     * @param manifest Other object to merge with.
+     */
+    default void addAllFiles(FileManifest manifest) {
```

Comment:
Sorry about the churn, but these methods were wedged in between the other overloads of `writeFile`. Me adding two new methods made this divide even worse, so I went ahead and moved these.